### PR TITLE
Fix race with query immediately following contract deployment

### PR DESCRIPTION
**Pull request by vistra**
RunQuery returns an error when the contract is not deployed. This will suppress the error when polling for the correct block height. If polling is unsuccessful the last error will be returned.

Another related fix - when submitting a transaction synchronously, the SDK treats HTTP 202 (indicating the transaction is still pending) as an error. Added this as an exception in the test flow.

### CODE REVIEW

**Comment by netoneko** [comment]
@OdedWx @electricmonk this is an issue that external developers will inevitably face, there needs to be a proper solution to this

**Comment by electricmonk** [comment]
@netoneko I agree and this is a long running argument I have with @OdedWx and @talkol. Sadly I'm currently on the losing side of this argument, maybe your experience with customers can help tip the needle.

File: test/e2e/harness.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ func (h *Harness) DeployContract(from *keys.Ed25519KeyPair, contractName string,
 	beginTime := time.Now()
 
 	txOut, txId, err := h.SendDeployTransaction(from.PublicKey(), from.PrivateKey(), contractName, processorType, code...)
-	if err != nil {
+	if err != nil && !strings.Contains(err.Error(), "http status 202 Accepted") { // TODO the SDK treats HTTP 202 as an error
```
Comment (electricmonk) [inline test/e2e/harness.go]: wrap in a function `isNonError`?

**Comment by vistra** [comment]
fixes #1328

**Pull request merged**